File: 2018/11/lastfactorialdigit/lastfactorialdigit.py
import sys
from functools import reduce

lines = sys.stdin.readlines()


def factorial(n):
    if n in [0, 1]:
        return 1
    else:
        return n * factorial(n - 1)


factorials = map(lambda s: s[-1], map(str, map(factorial, map(int, lines[1:]))))
output = '\n'.join(factorials)
print(output)

File: 2018/11/lastfactorialdigit/lastfactorialdigit.js
/*eslint no-console: "off"*/

const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false
});

const lines = [];

rl.on("line", line => {
  lines.push(line);
});

rl.on("close", () => {
  const factorials = lines
    .slice(1)
    .map(line => parseInt(line))
    .map(factorial)
    .map(String)
    .map(s => s.slice(-1));
  const output = factorials.join("\n");
  console.log(output);
});

function factorial(n) {
  if ([1, 1][n]) return [1, 1][n];
  return n * factorial(n - 1);
}
